libs/langchain/langchain/callbacks/tracers/evaluation.py

```diff
@@ -102,18 +104,20 @@ def _evaluate_in_project(self, run: Run, evaluator: RunEvaluator) -> None:
         """
         try:
             if self.project_name is None:
-                self.client.evaluate_run(run, evaluator)
+                feedback = self.client.evaluate_run(run, evaluator)
             with tracing_v2_enabled(
                 project_name=self.project_name, tags=["eval"], client=self.client
             ):
-                self.client.evaluate_run(run, evaluator)
+                feedback = self.client.evaluate_run(run, evaluator)
         except Exception as e:
             logger.error(
                 f"Error evaluating run {run.id} with "
                 f"{evaluator.__class__.__name__}: {e}",
                 exc_info=True,
             )
             raise e
+        example_id = str(run.reference_example_id)
+        self.logged_feedback.setdefault(example_id, []).append(feedback)
```

Comment:
maybe noob q but why can't feedback just be returned

Reply:
the callbacks are all called via the run manager in the depths of the callstack so it would just be ignored sadly